### DISCUSSION TITLE
不理解，为什么vb能直接用与key类型不一样的直接遍历

### DISCUSSION THREAD

**Discussion by qaqxhy** [Meloong-Git/PCL#3280] [提问] [answered]
VB:
```
Public AniGroups As New Dictionary(Of String, AniGroupEntry)
...
Dim i As Integer = -1
Dim Entry As AniGroupEntry = AniGroups.Values(i)
```
C#:
```
public Dictionary<string, AniGroupEntry> AniGroups = new();
...
int i = -1;
List<string> _keys = new(AniGroups.Keys);
AniGroupEntry entry = AniGroups[_keys[i]];
```

**Comment by SALTWOOD** [comment]
虽然不熟 VB，但是会不会和 Variant 类型的存在有关系（？

**Comment by Light-Beacon** [comment]
你这个例子里的
`AniGroups.Values(i)` (VB) 
不是和
`AniGroups.Values[i]` (C#)
一样吗，为什么要先把 keys 搞出来

**Reply by qaqxhy** [reply]
C#不行的，anigroups的key是string类型，所以如果i是string类型是可以这样用的，不然的话会提示类型问题
![image](https://github.com/Hex-Dragon/PCL2/assets/86876223/4ec874ce-cec6-4afe-9f68-2a31b4221e9a)

**Reply by Light-Beacon** [reply, marked as answer]
`AniGroups.Values.ElementAt(i)` 可以用这个
话说为什么不用 foreach （

**Reply by qaqxhy** [reply]
...
